add Label to LogMessage and colorize it

klyonrad opened the pull request:
This adds the level of the LogMessage as uppercase label and colorizes it when the level is WARN and above.

Inspired by the `:pretty` formatter

Closes #94

klyonrad commented:
@mattbrictson Hey, could you please help me out with that Test Suite? I can only get the tests passing for older SSHKit versions if I remove the color codes from the to asserted output.

mattbrictson commented:
Thanks for the PR! I am a bit busy and may not have time to review this right away. If there are any Airbrussh contributors/reviewers lurking on this thread, feel free to jump in. 😄

daniel-gomes-sociomantic commented:
would love to see this merged!

@klyonrad is there any way to try your solution locally?

klyonrad commented:
Ouf. Been a while since I coded that.

@daniel-gomes-sociomantic you can always apply the modified code in your project 😉 

Regarding the Pull Request, this requirement has to be satisfied:

> I'm willing to accept a PR that adds a colorized WARN, ERROR or FATAL prefix for those levels (leaving the log message itself un-colorized). No prefix or color for INFO.

which (judging from my own code comments) complied to

mattbrictson commented:
@daniel-gomes-sociomantic if you'd like to try this code in your project, specify airbrussh like this in your project's Gemfile:

```ruby
gem "airbrussh", :github => "klyonrad/airbrussh", :branch => "colorize-logmessage"
```

Then make sure to `bundle install` and run capistrano via:

```
bundle exec cap
```

ncreuschling commented:
I would like to see this incorporated.

mattbrictson commented on lib/airbrussh/console_formatter.rb:
```diff
@@ -69,7 +73,12 @@ def write(obj)
+      return print_indented_line(log_message.to_s) if verbosity < 2
+      # use level string to call corresponding colorize method
+      level = send(LEVEL_COLORS[verbosity], LEVEL_NAMES[verbosity].ljust(6))
+      print_indented_line(level + log_message.to_s)
```
The behavior looks good, but I think this code is a bit fragile because it depends on the verbosity levels being certain integer values. For example `if verbosity < 2` is not very clear when it could be written as `if verbosity < SSHKit::Logger::WARN`.

What do you think about using a `case` statement instead? Like this:

```ruby
def write_log_message(log_message)
  return if debug?(log_message)
  print_task_if_changed
  print_indented_line(format_log_message(log_message))
end

def format_log_message(log_message)
  case log_message.verbosity
  when SSHKit::Logger::WARN
    "WARN  #{yellow(log_message.to_s)}"
  when SSHKit::Logger::ERROR
    "ERROR #{red(log_message.to_s)}"
  when SSHKit::Logger::FATAL
    "FATAL #{red(log_message.to_s)}"
  else
    log_message.to_s
  end
end
```

klyonrad replied:
well it's more clear but needs more code lines 😉 

gonna do it with that

mattbrictson commented:
> Hey, could you please help me out with that Test Suite? I can only get the tests passing for older SSHKit versions if I remove the color codes from the to asserted output.

Actually the tests look good. If you are running Ruby 2.4, you'll get unrelated failures for older SSHKit versions because they are incompatible with 2.4. The Travis build is passing, which is what matters.

If you could rebase on `master` and address the implementation comment above, I will be happy to merge. Thanks!

klyonrad commented:
> Actually the tests look good. If you are running Ruby 2.4, you'll get unrelated failures for older SSHKit versions because they are incompatible with 2.4. The Travis build is passing, which is what matters.

Well, yeah the test worked because I removed checking the color 😉 
I reverted that, so you can look into the travis output, see https://travis-ci.org/mattbrictson/airbrussh/builds/220481968

But that revert commit can also be deleted again

mattbrictson commented:
Ah, sorry I missed that. Do you mind if I add a commit to your branch as an experiment to see if I can get the tests working?

klyonrad commented:
Do as you please! :)

mattbrictson commented:
OK, take a look at 4163da1 and let me know what you think!

klyonrad commented:
looks good, I deleted two middle commits and then force-pushed; so now it's one commit from and one from you :)

mattbrictson commented:
Great, I'll merge it in. Thanks for the PR and for your patience!
